pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration == nil || *pdb.Status.ObservedGeneration != pdb.Generation {
```

Comment:
There is nothing that bumps Generation since there is no spec change happening so it will always be 1 for PDBs. You will either need to bump Generation here when you do the update or you should probably use a different pair of Generation/ObservedGeneration because Generation is meant to be bumped only on spec changes.